runtime/unix.c

```diff
@@ -440,6 +440,16 @@ int caml_num_rows_fd(int fd)
 #endif
 }
 
+void caml_thread_getname(char* name)
+{
+  pthread_t self = pthread_self();
+#if defined(__OpenBSD__) || defined(__FreeBSD__)
+  pthread_get_name_np(self, name, MAX_THREAD_NAME_LENGTH);
+#else /* linux glibc/musl or NetBSD or apple */
+  pthread_getname_np(self, name, MAX_THREAD_NAME_LENGTH);
```

Comment:
it looks like Apple returns a single space character from `pthread_getname_np` when there has not been a name set before. Funny. I'm investigating.